proposals/infra/INF-0004-register-types-and-diagnostics.md

```diff
@@ -188,7 +188,16 @@ struct Eg9{
 };
 
 Eg9 e9;
-// error: location annotations cannot be specified on members
+// "error: location annotations cannot be specified on members"
+
+template<typename R>
+struct Eg10 {
+    R b;
+};
+
+Eg10<Texture2D> t : register(u0);
+// "warning: variable of type 'Eg10' bound to register type 'u' does not contain a matching 'UAV' resource"

```

Comment:
Does the requirement to look inside template expansions change anything about the plan on implementing this in handleHLSLResourceBindingAttr?

Reply:
`RequireCompleteType` needs to be called in template expansion cases, but there doesnt seem to be any fundamental reason why this specific location is insufficient for implementing this infrastructure.